### PR TITLE
Documentation improvements for eventing and autoscaling

### LINKED ISSUE #1923: Restructure "Eventing component" section
Align content with the developer flow and arrange content into logical sections.

[View proposed nav structure](https://docs.google.com/spreadsheets/d/1L3EAeTX7RRdZhcQZeNmxwN4DCncuImCXA5A1_KQvwSk/edit?ts=5d826691#gid=413198505&range=121:121)

- For all moved or renamed files, add a redirect to that file's frontmatter (#1925)
- [x] Create a Eventing only install section
- [ ] Create new “Event consumers” section
- [ ] Create new “Event producers” section
- [x] Create new “Debugging” section
- [x] Organize “Debugging (and Observabiltiy)” content into new section

### PR DESCRIPTION
**Pull request by abrennan89**
Fixes #1923 

## Proposed Changes

- Updated eventing README / landing to remove fluff, improve explanation of how eventing works
- Moved duck typing into concepts in developer guide
- Tweaks to autoscaler docs; concepts didn't explain any concepts so I renamed it to be more accurate
- Add def_list extension to mkdocs

### CODE REVIEW

**Review by n3wscott — APPROVED** [review]
/lgtm
/approve

File: docs/developer/eventing/sinks/README.md [hunk 1]
```diff
@@ -1,29 +1,12 @@
 # About sinks
 
-When you create an event source, you can specify a _sink_ where events are sent to from the source. A sink is an Addressable resource that can receive incoming events from other resources. Knative Services, Channels, and Brokers are all examples of sinks.
+When you create an event source, you can specify a _sink_ where events are sent to from the source. A sink is an _Addressable_ or a _Callable_ resource that can receive incoming events from other resources. Knative Services, Channels, and Brokers are all examples of sinks.
```
Comment (evankanderson) [inline docs/developer/eventing/sinks/README.md:3]: _Callable_ is intended to inherit from _Addressable_; every _Callable_ should be _Addressable_.

With that said, this seems fine.

File: docs/developer/eventing/sinks/README.md [hunk 2]
```diff
@@ -1,29 +1,12 @@
...
-      kind: KafkaSink
-      name: <kafka-sink-name>
-```
-
-Where;
-
-- `<trigger-name>` is the name of the Trigger being connected to the sink.
-- `<kafka-sink-name>` is the name of a KafkaSink object.
+Callable objects
+:   Callable objects are able to receive an event delivered over HTTP and transform the event, returning 0 or 1 new events in the HTTP response. These returned events may be further processed in the same way that events from an external event source are processed.
```
Comment (evankanderson) [inline docs/developer/eventing/sinks/README.md:9]: You want to either mention `status.address.url`, or mention that is an _Addressable_.

File: docs/eventing/README.md [hunk 3]
```diff
@@ -1,78 +1,37 @@
...
----
-
 # Knative Eventing
 
-Knative Eventing is a system that is designed to address a common need for cloud
-native development and provides composable primitives to enable late-binding
-event sources and event consumers.
+Knative Eventing enables developers to use an [event-driven architecture](https://en.wikipedia.org/wiki/Event-driven_architecture) with serverless applications. An event-driven architecture is based on the concept of decoupled relationships between event producers that create events, and event consumers, or [_sinks_](../../developer/eventing/sinks), that receive events.
+
+In a Knative Eventing deployment, event [Sources](../../eventing/sources) are the primary event producers, however you can also configure a sink or _subscriber_ to respond to HTTP requests by sending a response event. Examples of sinks in a Knative Eventing deployment include Knative Services, Channels and Brokers.
```
Comment (evankanderson) [inline docs/eventing/README.md:5]: Is it worth mentioning that existing software can also generate event by POSTing them to a Broker or Channel?

File: docs/eventing/README.md [hunk 4]
```diff
@@ -84,17 +43,4 @@ the registry contain (all) the required information for a consumer to create a
...
-together and for those use cases, Knative Eventing provides two additional
-resources:
-
-1. **[Sequence](./flows/sequence)** provides a way to define an in-order list of functions.
-1. **[Parallel](./flows/parallel)** provides a way to define a list of branches for events.
-
-## Observability
-
-- [Eventing Metrics API](../admin/collecting-metrics/eventing-metrics/metrics.md)
+[Event Registry documentation](../../eventing/event-registry).
```
Comment (devguyio) [inline docs/eventing/README.md:46]: this link as well, maybe `(./event-registry)` instead?

**Review by devguyio — COMMENTED** [review]
@abrennan89  Sorry if it's weird reviewing after the PR is merged, got merged mid-review. So I left my comments fwiw.

**Pull request merged**